### ISSUE TITLE
Make nodejs 10.11.0 (npm 6.4.1) strongly recommended (required?)

### ISSUE THREAD

**Issue by glunardi** [mattermost/mattermost-developer-documentation#350]
Currently, the [Developer Setup](https://developers.mattermost.com/contribute/mobile/developer-setup/) page reads:

`This includes NPM which is also needed. Currently version 10.11.0 is recommended with npm 6.4.1. 11.x is not working.`

After several hours of trying to create a reproducible build environment for our needs, I believe that the version 10.11.0 with npm 6.4.1 is not only recommended, it is required.

None of the official current releases of NodeJS build mattermost-mobile successfully.

I could not get any of these versions to work:
10.16.3 (npm 6.9.0) - memory leak around source-map
12.9.1 (npm 6.10.2) - problems with redux and deps
8.16.1 (npm 6.4.1) - can't remember the error(s)

That means that none of the nodejs docker images worked for us. I am not a technical user anymore so I could have missed something.

There is one easy way to prevent people from spending the time I did on this.

Would you be willing to consider making the assertion about nodejs and npm versions more strongly?

Possibly include that information in the mattermost-mobile project readme, update the Developer Setup page to make it clear that this is what Mattermost is using.

**Comment by haardikdharma10** [comment]
Hey, I would like to fix this. Also, any other suggestions you'd like to add? @hanzei?

**Comment by hanzei** [comment]
Thanks for taking a look @haardikdharma10. It would be nice to have an overview of the working nodejs versions. That would be 10, 12 and 14 right now. I think it's fine to test with the latest version of each major.

**Comment by haardikdharma10** [comment]
Thanks @hanzei, I am working on it. Also, the line `If you run into any issues getting your environment set up, check the Troubleshooting section at the bottom for common solutions.` I don't see any troubleshooting section at the bottom and the link: https://developers.mattermost.com/contribute/mobile/developer-setup/#troubleshooting doesn't make sense. Should we consider removing it? 🤔

**Comment by glunardi** [comment]
@haardikdharma10 thank you very much for looking into this. Just a few days ago I attempted to build a newer version again and failed to do so (limited time).

Some of the issues I very recently faced:
- Android's API version (and build-tools) seems to be quite recent (undocumented) 
- NodeJS (this ticket) version is difficult to nail down. Better documentation, recommendation would be great
- One project, eslint seemed incompatible with the 3 or 4 versions I have tried

Would love to help test drive any new documentation content. I have failed at building mattermost-mobile recently :-).
